### DISCUSSION TITLE
Getting biography field information.

### DISCUSSION THREAD

**Discussion by MustafaBKLZ** [diezo/Ensta#87] [Show and tell]
I encountered a problem like this. 

If the biography is written under each other with enter key, for example:

```
    bla bla
    bla bla
    bla bla
```

Then result is like this;
    `[1]=> string(7) "bla bla" [2]=> string(7) "bla bla" [3]=> string(7) "bla bla"`

However, if I write my biography information without enter;

    `bla bla bla bla bla bla`

then result is;
`[1]=> string(23) "bla bla bla bla bla bla"`

My question is this;
**How can we get all the biography information at once? Is there a way around this?**

I using like this.
```python
from ensta import Host
host = Host(username, password)
profile = host.profile("instagramusername")
print(profile.biography)
print(profile.biography_links)
print(profile.category_name)
print(profile.following_count)
```

**Comment by diezo** [comment]
seems like instagram has updated the biography api. i'll check what's happening.

**Reply by MustafaBKLZ** [reply]
Ok. Thanks. I'm waiting for your reply. :)

**Comment by ettoreleandrotognoli** [comment]
~~I think ensta is not using this endpoint~~
Sorry, it is using, it is in the Guest class
https://www.instagram.com/api/v1/users/web_profile_info/?username=dev.ettore

The biography is looking "normal", just one string with "\n".

```json
{
    "data": {
        "user": {
            "ai_agent_type": null,
            "biography": "Prof. @unimaroficial |  Mestre em Ciência da Computação - UFScar\n💻 Computação do básico ao avançado\n🤖 Transformando conceitos simples em magia com IA",
            "bio_links": [],
            "fb_profile_biolink": null,
            "biography_with_entities": {
                "raw_text": "Prof. @unimaroficial |  Mestre em Ciência da Computação - UFScar\n💻 Computação do básico ao avançado\n🤖 Transformando conceitos simples em magia com IA",
                "entities": [
                    {
                        "user": {
                            "username": "unimaroficial"
                        },
                        "hashtag": null
                    }
                ]
            },
            ...
```

**Reply by ettoreleandrotognoli** [reply]
The code snippet is working normally for me too.

```python
from ensta import Host
host = Host(username, password)
profile = host.profile("instagramusername")
print(profile.biography)
print(profile.biography_links)
print(profile.category_name)
print(profile.following_count)
```

**Comment by MustafaBKLZ** [comment]
Hello,
Now I tried a lot of things and realized that I was missing some information. Still I couldn't find out why I was having the above issue.
I use this library with php.

``` php
 $username = "mustafabukulmezcom";
 $python = 'C:\\Users\\Mustafa\\AppData\\Local\\Programs\\Python\\Python310\\python.exe'; 
 $pyscript = "C:\\xampp\\htdocs\\libraries\\py\\insta_read.py " . $username ;
 exec("$python $pyscript", $output, $return);
 var_dump($output);
```

First of all, my Python codes were like this.
```python
from ensta import Host
host = Host(username, password)
profile = host.profile("instagramusername")
print(profile.biography)
print(profile.biography_links)
print(profile.category_name)
print(profile.following_count)
```

When I use it this way, the output I get on the PHP side is like this. Index 2,3,4 all my biography @mustafabukulmezcom
```
array(25) { 
          [0]=> string(18) "mustafabukulmezcom" 
          [1]=> string(18) "Software Developer" 
          [2]=> string(15) "@dijitalortagim" 
          [3]=> string(44) "Content Creator, Sharing, Coding, Technology"  ......
```

I started this discussion because of the problem so far. However, when I tried more, I realized that I could not get any information from some profiles. Ex: @design
```
array(0) { }
```


I realized something was wrong here. Afterwards, I edited Python codes.
```python

#!/usr/bin/env python
# to pass parameters to this file 
import sys

# read parameter
who = sys.argv[1]

from ensta import Host
host = Host("username", 'password')
profile = host.profile(who)

# Organize data into an associative array
data = {
    "Username": who,
    "Biography": profile.biography,
    "Biography Links": profile.biography_links,
    "Category Name": profile.category_name,
    "Following Count": profile.following_count,
    "Followed by Viewer": profile.followed_by_viewer,
    "Follower Count": profile.follower_count,
    "Full Name": profile.full_name,
......
}

# Convert data to JSON format
import json
json_data = json.dumps(data)

# Print JSON data
print(json_data)
```

Now I can get information from their profiles, which I just noticed was missing information. @design
```
array(1) { 
          [0]=> string(1291) "{
                "Username": "design", 
                "Biography": "Discover design through the eyes of creators \u2728", "Biography Links": [], 
                "Category Name": "App page", 
                "Following Count": 2032, 
                "Followed by Viewer": true, 
                "Follower Count": 2164220, 
                "Full Name": "Instagram\u2019s @design", 
                "Has Channel": false, 
                "Highlight Count": 18, " ..................
```

However, the problem that caused this discussion was not seen in this way. Here; @mustafabukulmezcom
```
array(1) { 
          [0]=> string(1612) "{
                "Username": "mustafabukulmezcom", 
                "Biography": "Software Developer \n@dijitalortagim \nContent Creator, Sharing, Coding, Technology", 
                ...................
```



I gave all this information so maybe it will make it easier for you to find out what's wrong. Perhaps it will become clear that trying further is unnecessary.

**Reply by ettoreleandrotognoli** [reply]
I understood that before you start to use json, your PHP code was using line separators as field separators, so the "Biography" with line separators in it was badly decoded
The `json.dumps(data)` will not use line breaks/separators by default, so it is a good approach for what you are doing
But in that case there is nothing wrong with the ensta

**Reply by MustafaBKLZ** [reply]
Yes, that's why I wanted to write in detail as soon as I noticed the incident.

**Reply by diezo** [reply]
@MustafaBKLZ Appreciate the detailed explanation. now that you've found the solution, I'll close this discussion now.

**Reply by diezo** [reply]
And yes, if ensta has added value to your project, don't forget to give this repo a star. Thanks.

**Reply by MustafaBKLZ** [reply]
Okey you can close it. Thanks all help and answers.
